### PR TITLE
Close #18845: Adds swipe-to-delete to tabs tray refactor 

### LINKED ISSUE #18845: Add swipe to delete on browser tray lists
We need to observe when we're in multi-select mode so that we can remove the `TabTouchCallback` when it's enabled.

### PR DESCRIPTION
**Pull request by jonalmeida**
Copied the `TabsTouchHelper` from the `tabtray` package here so we don't
need to re-write our own because there's nothing more to add.

We can hook this up with our tabs tray here by putting it in the
`BaseBrowserTrayList` for our normal and private tabs list.

Unrelated: fixed a bug/typo in `BrowserTabsAdapter` as well.

https://user-images.githubusercontent.com/1370580/113968477-29497000-9801-11eb-8727-195303485ed8.mp4

Fixes #18845 

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabstray/NavigationInteractor.kt [hunk 1]
```diff
@@ -14,8 +14,7 @@ import org.mozilla.fenix.components.metrics.MetricController
 import org.mozilla.fenix.home.HomeFragment
 
 /**
- * For interacting with UI that is specifically for [TabsTrayFragment[]] and other browser
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/NavigationInteractor.kt:17]: This kdoc didn't make sense when working on this PR, so I corrected here. 😄

File: app/src/main/java/org/mozilla/fenix/tabstray/browser/TabsTouchHelper.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        dX: Float,
+        dY: Float,
+        actionState: Int,
+        isCurrentlyActive: Boolean
+    ) {
+        super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive)
+
+        if (recyclerView.context.settings().gridTabView) {
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/browser/TabsTouchHelper.kt:72]: I took this code from the old package, but I think we can refactor this bit out.

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabstray/browser/TabsTouchHelper.kt]: This is done. We now have a `onViewHolderDraw` callback that delegates it to the app.

**Review by rocketsroger — APPROVED** [review]
LGTM!

**Comment by jonalmeida** [comment]
Had to fix the tests to correct the arguments in the constructor.

**Pull request merged**